src/components/dataTable/js/tableCellDirective.js

```diff
@@ -0,0 +1,213 @@
+angular
+  .module('material.components.table')
+  .directive('mdCell', MdTableCellDirective)
+  .directive('mdAlign', MdTableAlignDirective)
+  .directive('mdOrderBy', MdTableOrderByDirective);
+
+
+function MdTableCellDirective() {
+
+  return {
+    restrict: 'A',
+    require: ['mdCell', '^^mdTable'],
+    controller: MdTableCellController,
+    controllerAs: '$mdCell',
+    link: postLink
+  };
+
+  function postLink(scope, element, attrs, ctrls) {
+    var ctrl = ctrls[0];
+    var tableCtrl = ctrls[1];
+
+    ctrl.initCtrl(tableCtrl);
+  }
+}
+
+function MdTableCellController($element, $scope) {
+  this.$element = $element;
+  this.$scope = $scope;
+
+  this.validAlignments = ['left', 'center', 'right', 'justify'];
+}
+
+MdTableCellController.prototype.initCtrl = function(tableCtrl) {
+  this.table = tableCtrl;
+
+  this.$scope.$watch(this.getAlignment.bind(this), function(newAlignment, oldAlignment) {
+
+    if (this.validAlignments.indexOf(newAlignment) !== -1) {
+      this.$element.addClass('md-align-' + newAlignment);
+    }
+
+    if (this.validAlignments.indexOf(oldAlignment) !== -1 && newAlignment !== oldAlignment) {
+      this.$element.removeClass('md-align-' + oldAlignment);
+    }
+
+  }.bind(this));
+};
+
+MdTableCellController.prototype.getAlignment = function() {
+
+  // Transform NodeList into an Array.
+  var headRows = [].slice.call(this.table.getHead().rows);
+  var cellIndex = this.$element[0].cellIndex;
+
+  return headRows
+    .map(function(row) {
+      return angular.element(row.cells[cellIndex]).data('align');
+    })
+    .filter(function(cell) {
+      return !!cell;
+    })[0];
+};
+
+function MdTableAlignDirective() {
+
+  return  {
+    restrict: 'A',
+    require: ['mdCell', '^^mdHead'],
+    link: postLink
+  };
+
+  function postLink(scope, element, attrs) {
+    attrs.$observe('mdAlign', function(align) {
+      element.data('align', align);
+    });
+  }
+}
+
+function MdTableOrderByDirective() {
+
+  return {
+    restrict: 'A',
+    require: ['mdOrderBy', 'mdCell', '^^mdOrder', '^^mdTable'],
+    scope: {},
+    controller: MdTableOrderByController,
+    controllerAs: '$mdOrderBy',
+    link: postLink
+  };
+
+  function postLink(scope, element, attrs, ctrls) {
+    var ctrl = ctrls[0];
+    var cellCtrl = ctrls[1];
+    var orderCtrl = ctrls[2];
+    var tableCtrl = ctrls[3];
+
+    ctrl.initCtrl(tableCtrl, orderCtrl, cellCtrl);
+  }
+}
+
+function MdTableOrderByController($element, $scope, $attrs, $compile, $mdUtil, $$mdSvgRegistry) {
+  this.$element = $element;
+  this.$scope = $scope;
+  this.$attrs = $attrs;
+  this.$compile = $compile;
+  this.$mdUtil = $mdUtil;
+  this.$$mdArrowUpIcon = $$mdSvgRegistry.mdArrowUp;
+
+  this.isActiveRegex = new RegExp('^-?' + this.$attrs.mdOrderBy + '$');
```

Comment:
There would be no benefit from it. It would just break the Prototype Pattern. Having it here is the right place, since this is called only once as same as a constructor.